state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2098,7 +2102,7 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			let new_boost_status = if delay > Default::default() {
 				let process_at_block = frame_system::Pallet::<T>::block_number() + delay;
 				PendingPrewitnessedDeposits::<T, I>::append(process_at_block, deposit);
-				BoostStatus::BoostPending
+				BoostStatus::BoostPending { amount }
```

Comment:
Is the amount enough to distinguish this? 

Maybe it would be better to assign the `prewitness_deposit_id` and use that instead?

Reply:
I think i get it: as long as the amounts match, we can consider them 'fungible' in a sense. We should think about how this works alongside the assumptions in tx screening: imagine there are two txs with matching amounts, but one is flagged for rejection. Does this create any problems?